File: src/main/java/core/file/xml/UnsupportedHattrickDataVersionException.java
package core.file.xml;

import lombok.Getter;

@Getter
class UnsupportedHattrickDataVersionException extends XMLParseException {

    private final String fileName;
    private final String expectedVersion;
    private final String actualVersion;

    public UnsupportedHattrickDataVersionException(String fileName, String expectedVersion, String actualVersion) {
        super("Unsupported Hattrick Data Version: '%s' (expected '%s' for filename '%s')"
            .formatted(actualVersion, expectedVersion, fileName));
        this.fileName = fileName;
        this.expectedVersion = expectedVersion;
        this.actualVersion = actualVersion;
    }
}

File: src/main/java/core/file/xml/UnsupportedHattrickDataVersionException.kt
package core.file.xml

internal class UnsupportedHattrickDataVersionException(
    val fileName: String?,
    val expectedVersion: String?,
    val actualVersion: String?
) : XMLParseException(
    String.format(
        "Unsupported Hattrick Data Version: '%s' (expected '%s' for filename '%s')",
        actualVersion, expectedVersion, fileName
    )
)
